controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -646,19 +637,54 @@ private Scope getScope(final String scopeName) {
 
     private CompletableFuture<Long> findNumMerges(String scopeName, String streamName, Executor executor) {
         return getScaleMetadata(scopeName, streamName, null, executor).thenApply(scaleMetadataList -> {
-            long size = scaleMetadataList.size();
+            int size = scaleMetadataList.size();
```

Comment:
both these methods have a lot of duplication.. you can consolidate them into a single helper method and call that..